packages/xstate-fsm/test/fsm.test.ts

```diff
@@ -258,4 +258,44 @@ describe('interpreter', () => {
 
     expect(executed).toBe(true);
   });
+
+  it('should reveal the current state', () => {
+    const machine = createMachine({
+      initial: 'test',
+      context: { foo: 'bar' },
+      states: {
+        test: {}
+      }
+    });
+    const service = interpret(machine);
+
+    service.start();
+
+    expect(service.state.value).toEqual('test');
+    expect(service.state.context).toEqual({ foo: 'bar' });
+  });
+
+  it('should reveal the current state after transition', done => {
+    const machine = createMachine({
+      initial: 'test',
+      context: { foo: 'bar' },
+      states: {
+        test: {
+          on: { CHANGE: 'success' }
+        },
+        success: {}
+      }
+    });
+    const service = interpret(machine);
+
+    service.start();
+
+    service.subscribe(() => {
```

Comment:
does this have to be async? the test would be pretty much the same if you would just assert this after `service.send('CHANGE');`